### ISSUE TITLE
Prophecy 2.0

### ISSUE THREAD

**Issue by theolepage** [theolepage/prophecy#35]
- [ ] Simple tensor class
- [ ] Layers: dense, relu, sigmoid, softmax
- [ ] Model: predict, fit (validation), evaluate, summary
- [ ] Loss: MSE, cross-entropy
- [ ] Metrics: loss, accuracy
- [ ] Weights regularization
- [ ] Learning rate schedulers
- [ ] Optim: Adam, SGD
- [ ] Python bindings
- [ ] Export/import weights
- [ ] Optmizations: CUDA, OpenMP
- [ ] Clean CPP, cmake, tests, docs, example notebooks

Objective: fast and accurate classification of MNIST.
